extensions/parquet/table/src/main/java/io/deephaven/parquet/table/metadata/RowGroupInfo.java

```diff
@@ -0,0 +1,352 @@
+        }
+
+        @Override
+        public boolean hasNext() {
+            return nextIter < numRowGroups;
+        }
+
+        @Override
+        public Table next() {
+            try {
+                final RowSet rawRowSet = input.getRowSet();
+
+                final long nextSz = impliedRowGroupSz + (nextIter < fractionalGroups ? 1 : 0);
+                final WritableRowSet nextRows = rawRowSet.subSetByPositionRange(startOffset, startOffset += nextSz);
+                return input.getSubTable(nextRows.toTracking());
+            } finally {
+                nextIter++;
+            }
+        }
+    }
+
+    /**
+     * Splits evenly across a number of RowGroups, ensuring that no group is larger than {@code maxRows}
+     */
+    private static class SplitByMaxRows extends IterativeRowGroupInfo {
+        private final long maxRows;
+
+        private SplitByMaxRows(long maxRows) {
+            if (maxRows <= 0) {
+                throw new IllegalArgumentException("MaxRows must be positive");
+            }
+            this.maxRows = maxRows;
+        }
+
+        @Override
+        public Iterator<Table> getIterator(final @NotNull Table input) {
+            final long numRowGroups = (input.size() / maxRows) + (input.size() % maxRows > 0 ? 1 : 0);
+            if (numRowGroups == 1) {
+                return DEFAULT.getIterator(input);
+            } else {
+                return new SplitEvenly(numRowGroups).getIterator(input);
+            }
+        }
+
+        @Override
+        public boolean hasNext() {
+            throw new UnsupportedOperationException("SplitByMaxRows.hasNext() not implemented");
+        }
+
+        @Override
+        public Table next() {
+            throw new UnsupportedOperationException("SplitByMaxRows.next() not implemented");
+        }
+    }
+
+    /**
+     * Splits into RowGroups based on unique values within {@code groups}, and optionally further splits (per unique
+     * {@code groups} ensuring that no group is larger than {@code maxRows}. If {@code maxRows} is not desired, this
+     * parameter may be set to {@code Long.MAX_VALUE}
+     */
+    private static class SplitByGroups extends IterativeRowGroupInfo {
+        private final long maxRows;
+        private final String[] groups;
+        private final IterativeRowGroupInfo subSplitInfo;
+
+        private Table[] partitionedTables;
+        private int nextTable;
+
+        private Iterator<Table> subIter;
+
+        private SplitByGroups(long maxRows, final @NotNull String[] groups) {
+            this.maxRows = maxRows;
+            this.groups = groups;
+            this.subSplitInfo = maxRows == Long.MAX_VALUE ? DEFAULT : new SplitByMaxRows(maxRows);
+        }
+
+        @Override
+        public Iterator<Table> getIterator(final @NotNull Table input) {
+            final Table[] newPartitions = input.partitionBy(groups).constituents();
+            ensureConsistentOrdering(input, newPartitions);
```

Comment:
This is not something we can depend on, we don't know that the keyspace of the resulting tables is the same as the input table.